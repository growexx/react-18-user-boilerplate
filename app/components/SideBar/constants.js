/*
 * Sidebar Constants
 * This file contains constants used in Sidebar component.
 */

import React from 'react';
import {
  UserOutlined,
  SmileOutlined,
  LoadingOutlined,
  ExportOutlined,
  NumberOutlined,
  FormOutlined,
  LockOutlined,
  PieChartOutlined,
  MessageOutlined,
  ShoppingOutlined,
  SearchOutlined,
  TableOutlined,
} from '@ant-design/icons';
import {
  ROUTES,
  ROLE_BASED_SIDEBAR_MENU,
  ROLES,
  RESTRICTED_ROUTES,
} from 'containers/constants';

export const MenuItems = [
  {
    to: ROUTES.HOME,
    tabName: 'Home',
    icon: <SmileOutlined />,
  },
  {
    to: ROUTES.LOADER,
    tabName: 'Loader Demo',
    icon: <LoadingOutlined />,
  },
  {
    to: ROUTES.EXPORT_DATA,
    tabName: 'Export Data To CSV',
    icon: <ExportOutlined />,
  },
  {
    to: ROUTES.NUMERAL_CONVERTER,
    tabName: 'Number Conversion Demo',
    icon: <NumberOutlined />,
  },
  {
    to: ROUTES.USERS,
    tabName: 'User Management',
    icon: <UserOutlined />,
  },
  {
    to: ROUTES.GITHUB_SEARCH,
    tabName: 'Github Search',
    icon: <SearchOutlined />,
  },
  {
    to: ROUTES.TEST_ADMIN_PAGE,
    tabName: 'Admin Page',
    icon: <LockOutlined />,
  },
  {
    to: ROUTES.CHARTS,
    tabName: 'Charts',
    icon: <PieChartOutlined />,
  },
  {
    to: ROUTES.MULTI_TAB_SUPPORT,
    tabName: 'Multi Tab Communication',
    icon: <MessageOutlined />,
  },
  {
    to: ROUTES.PRODUCTS,
    tabName: 'Products',
    icon: <ShoppingOutlined />,
  },
  {
    to: ROUTES.REACT_HOOK_FORM,
    tabName: 'React Hook Form',
    icon: <FormOutlined />,
  },
  {
    to: ROUTES.AG_GRID,
    tabName: 'AG Grid',
    icon: <TableOutlined />,
  },
];

/**
 * Filters Sidebar menu based on role
 * @param {string} role
 */
export const GET_FILTERED_MENU_ITEM = role =>
  MenuItems.filter(item =>
    // Check if route is private then role has access to it
    RESTRICTED_ROUTES.includes(item.to)
      ? ROLE_BASED_SIDEBAR_MENU[role || ROLES.USER].includes(item.to)
      : true,
  );

// component name as ga label
export const GA_LABEL_SIDEBAR = 'SideBar';
