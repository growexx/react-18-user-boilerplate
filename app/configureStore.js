import { configureStore } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import { createReduxHistoryContext } from 'redux-first-history';

import { repoApi } from './containers/HomePage/reposApiSlice';
import { appSlice } from './containers/App/slice';
import { apiSlice } from './apiSlice';
import languageSlice from './containers/LanguageProvider/slice';
import loginSlice from './containers/Auth/Login/slice';
import { authApi } from './containers/Auth/Login/authApiSlice';
import twoFactorAuthSlice from './containers/Auth/TwoFactorAuthentication/slice';

const { createReduxHistory, routerMiddleware, routerReducer } =
  createReduxHistoryContext({ history: createBrowserHistory() });

const store = configureStore({
  reducer: {
    router: routerReducer,
    app: appSlice,
    language: languageSlice,
    login: loginSlice,
    twoFactorAuth: twoFactorAuthSlice,
    repos: repoApi.reducer,
    auth: authApi.reducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware(),
    apiSlice.middleware,
    routerMiddleware,
  ],
  devTools: process.env.NODE_ENV !== 'production',
});

const history = createReduxHistory(store);

export { store, history };
