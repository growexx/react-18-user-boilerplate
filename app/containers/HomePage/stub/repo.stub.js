export const MOCK_REPO_RESPONSE = {
  id: 11116372,
  node_id: 'MDEwOlJlcG9zaXRvcnkxMTExNjM3Mg==',
  name: 'keystone-classic',
  full_name: 'keystonejs/keystone-classic',
  private: false,
  owner: {
    login: 'keystonejs',
    id: 6118534,
    node_id: 'MDEyOk9yZ2FuaXphdGlvbjYxMTg1MzQ=',
    avatar_url: 'dummy_url',
    gravatar_id: '',
    url: 'dummy_url',
    html_url: 'dummy_url',
    followers_url: 'dummy_url',
    following_url: 'dummy_url',
    gists_url: 'dummy_url',
    starred_url: 'dummy_url',
    subscriptions_url: 'dummy_url',
    organizations_url: 'dummy_url',
    repos_url: 'dummy_url',
    events_url: 'dummy_url',
    received_events_url: 'dummy_url',
    type: 'Organization',
    site_admin: false,
  },
  html_url: 'dummy_url',
  description: 'Node.js CMS and web app framework',
  fork: false,
  url: 'dummy_url',
  forks_url: 'dummy_url',
  keys_url: 'dummy_url',
  collaborators_url: 'dummy_url',
  teams_url: 'dummy_url',
  hooks_url: 'dummy_url',
  issue_events_url: 'dummy_url',
  events_url: 'dummy_url',
  assignees_url: 'dummy_url',
  branches_url: 'dummy_url',
  tags_url: 'dummy_url',
  blobs_url: 'dummy_url',
  git_tags_url: 'dummy_url',
  git_refs_url: 'dummy_url',
  trees_url: 'dummy_url',
  statuses_url: 'dummy_url',
  languages_url: 'dummy_url',
  stargazers_url: 'dummy_url',
  contributors_url: 'dummy_url',
  subscribers_url: 'dummy_url',
  subscription_url: 'dummy_url',
  commits_url: 'dummy_url',
  git_commits_url: 'dummy_url',
  comments_url: 'dummy_url',
  issue_comment_url: 'dummy_url',
  contents_url: 'dummy_url',
  compare_url: 'dummy_url',
  merges_url: 'dummy_url',
  archive_url: 'dummy_url',
  downloads_url: 'dummy_url',
  issues_url: 'dummy_url',
  pulls_url: 'dummy_url',
  milestones_url: 'dummy_url',
  notifications_url: 'dummy_url',
  labels_url: 'dummy_url',
  releases_url: 'dummy_url',
  deployments_url: 'dummy_url',
  created_at: '2013-07-02T05:01:55Z',
  updated_at: '2023-10-27T05:49:50Z',
  pushed_at: '2023-07-28T10:11:43Z',
  git_url: 'dummy_url',
  ssh_url: 'dummy_url',
  clone_url: 'dummy_url',
  svn_url: 'dummy_url',
  homepage: 'https://v4.keystonejs.com',
  size: 151918,
  stargazers_count: 14667,
  watchers_count: 14667,
  language: 'JavaScript',
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: true,
  has_discussions: false,
  forks_count: 2287,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 359,
  license: {
    key: 'mit',
    name: 'MIT License',
    spdx_id: 'MIT',
    url: 'dummy_url',
    node_id: 'MDc6TGljZW5zZTEz',
  },
  allow_forking: true,
  is_template: false,
  web_commit_signoff_required: false,
  topics: ['cms', 'javascript', 'keystonejs', 'nodejs'],
  visibility: 'public',
  forks: 2287,
  open_issues: 359,
  watchers: 14667,
  default_branch: 'master',
};
