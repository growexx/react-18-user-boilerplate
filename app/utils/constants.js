export const RESTART_ON_REMOUNT = '@@saga-injector/restart-on-remount';
export const DAEMON = '@@saga-injector/daemon';
export const ONCE_TILL_UNMOUNT = '@@saga-injector/once-till-unmount';
export const TOKEN_KEY = 'jwtToken';
export const USER_DATA_KEY = 'userData';
export const EMITTER_EVENTS = {
  LOG_IN: 'USER_LOGGED_IN',
  LOG_OUT: 'USER_LOGGED_OUT',
};
export const GA_CATEGORY_MENU_CLICKS = 'Menu_Clicks';
export const BASE_URL = 'https://api.github.com';
export const PRIMARY_COLOR = '#4d186e';
export const PRIMARY_HOVER = '#3E1358';
export const ERROR_COLOR = '#ff0000';
export const BASE_COLOR = '#000';
export const FONT_FAMILY =
  "'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif";
export const themeConfig = {
  token: {
    colorPrimary: PRIMARY_COLOR,
    colorPrimaryHover: PRIMARY_HOVER,
    colorError: ERROR_COLOR,
    colorTextBase: BASE_COLOR,
    colorLink: PRIMARY_COLOR,
    fontFamily: FONT_FAMILY,
  },
};
