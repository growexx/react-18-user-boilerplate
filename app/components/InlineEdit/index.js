/**
 *
 * InlineEdit
 *
 */

import React, { memo, useState, useEffect, useRef } from 'react';
import { Input, Button } from 'antd';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck, faTimes } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import { ESC_KEY_CODE } from 'components/InlineEdit/constants';
import { StyledInlineInput } from 'components/InlineEdit/StyledInlineInput';
import { TEST_IDS } from 'components/InlineEdit/stub';

function InlineEdit ({ value, onSave, placeholder }) {
  const [isInputActive, setIsInputActive] = useState(false);
  const [inputValue, setInputValue] = useState(value);
  const inputContainerRef = useRef(null);

  const handleOutSide = e => {
    if (
      inputContainerRef &&
      inputContainerRef.current &&
      inputContainerRef.current.contains(e.target)
    ) {
      return;
    }
    if (isInputActive) {
      handleSave();
    }
  };

  useEffect(() => {
    document.addEventListener('mousedown', handleOutSide);
    document.addEventListener('touchstart', handleOutSide);

    return () => {
      document.removeEventListener('mousedown', handleOutSide);
      document.removeEventListener('touchstart', handleOutSide);
    };
  }, []);

  const handleSave = () => {
    onSave(inputValue);
    setIsInputActive(false);
  }

  const onCancel = () => {
    setInputValue(value);
    setIsInputActive(false);
  };

  const handleKeyDown = e => {
    if (e.keyCode === ESC_KEY_CODE) {
      onCancel();
    }
  };

  const handleOnChange = e => {
    setInputValue(e.target.value);
  };

  const handleDoubleClick = () => {
    setIsInputActive(true);
  };

  const saveIcon = () => (
    <Button type="link">
      <FontAwesomeIcon
        icon={faCheck}
        onClick={handleSave}
        data-testid={TEST_IDS.SAVE_BUTTON}
        title={TEST_IDS.SAVE_BUTTON}
      />
    </Button>
  );

  const cancelIcon = () => (
    <Button type="link">
      <FontAwesomeIcon
        icon={faTimes}
        onClick={onCancel}
        data-testid={TEST_IDS.CANCEL_BUTTON}
        title={TEST_IDS.CANCEL_BUTTON}
      />
    </Button>
  );

  const addonAfterView = () => (
    <span>
      {saveIcon()}
      {cancelIcon()}
    </span>
  );

  const getValue = () => {
    if (value.length > 0) {
      return value;
    }
    return <span className="placeHolderView">{placeholder}</span>;
  };

  return (
    <StyledInlineInput ref={inputContainerRef}>
      {!isInputActive ? (
        <div
          data-testid={TEST_IDS.INPUT_VALUE}
          onDoubleClick={handleDoubleClick}
          className="inputValue"
        >
          {getValue()}
        </div>
      ) : (
        <Input
          placeholder={placeholder}
          suffix={addonAfterView()}
          value={inputValue}
          onChange={handleOnChange}
          allowClear
          onPressEnter={handleSave}
          onKeyDown={handleKeyDown}
          data-testid={TEST_IDS.INPUT_EDIT}
        />
      )}
    </StyledInlineInput>
  );
  
}

InlineEdit.propTypes = {
  value: PropTypes.string,
  onSave: PropTypes.func,
  placeholder: PropTypes.string,
};

export default memo(InlineEdit);
