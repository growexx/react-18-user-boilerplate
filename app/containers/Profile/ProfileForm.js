/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/anchor-is-valid */
/**
 *
 * ProfileForm Form
 *
 */

import React, { useState } from 'react';
import { EditorState } from 'draft-js';
import { FormattedMessage } from 'react-intl';
import parse from 'html-react-parser';
import { Helmet } from 'react-helmet';
import { Card, Button } from 'antd';
import { stateToHTML } from 'draft-js-export-html';
import FileUpload from 'components/FileUpload/Loadable';
import InlineEdit from 'components/InlineEdit';
import { CardExtraContainer } from './StyledProfile';
import { DATA_TEST_IDS, PROFILE_PLACEHOLDER } from './constants';
import { options } from './helper';
import messages from './messages';
import RichTextEditor from '../../components/RichTextEditor';

function ProfileForm() {
  const [nameContent, setNameContent] = useState('John Doe');
  const [designationContent, setDesignationContent] = useState(
    'Software Engineer at GrowExx',
  );
  const [locationContent, setLocationContent] = useState('Ahmedabad, India');
  const [aboutContent, setAboutContent] = useState(EditorState.createEmpty());
  const [experienceContent, setExperienceContent] = useState(
    EditorState.createEmpty(),
  );
  const [educationContent, setEducationContent] = useState(
    EditorState.createEmpty(),
  );
  const [
    licensesAndCertificationsContent,
    setLicensesAndCertificationsContent,
  ] = useState(EditorState.createEmpty());
  const [editAbout, setEditAbout] = useState(false);
  const [editExperience, setEditExperience] = useState(false);
  const [editEducation, setEditEducation] = useState(false);
  const [editLicensesAndCertifications, setEditLicensesAndCertifications] =
    useState(false);

  const isContentEdited = state =>
    state.getCurrentContent().getPlainText('\u0001').length > 0
      ? stateToHTML(state.getCurrentContent(), options)
      : PROFILE_PLACEHOLDER;

  return (
    <div>
      <Helmet>
        <title>Profile Form</title>
        <meta name="description" content="Description of ProfileForm" />
      </Helmet>
      <div className="u-mb-1">
        <InlineEdit
          value={nameContent}
          onSave={value => setNameContent(value)}
          placeholder="Enter Name"
        />
      </div>
      <div className="u-mb-1">
        <InlineEdit
          value={designationContent}
          onSave={value => setDesignationContent(value)}
          placeholder="Enter Designation"
        />
      </div>
      <div className="u-mb-5">
        <InlineEdit
          value={locationContent}
          onSave={value => setLocationContent(value)}
          placeholder="Enter Location"
        />
      </div>
      <Card
        hoverable
        type="inner"
        title="About"
        extra={
          <CardExtraContainer>
            {editAbout ? (
              <Button
                data-testid={DATA_TEST_IDS.ABOUT_SAVE}
                onClick={() => {
                  setEditAbout(false);
                }}
              >
                <FormattedMessage {...messages.save} />
              </Button>
            ) : (
              <Button
                data-testid={DATA_TEST_IDS.ABOUT_EDIT}
                onClick={() => {
                  setEditAbout(true);
                }}
              >
                <FormattedMessage {...messages.edit} />
              </Button>
            )}
          </CardExtraContainer>
        }
      >
        {editAbout ? (
          <RichTextEditor
            testId={DATA_TEST_IDS.ABOUT_EDITOR}
            value={aboutContent}
            onChange={value => {
              setAboutContent(value);
            }}
          />
        ) : (
          parse(`${isContentEdited(aboutContent)}`)
        )}
      </Card>
      <br />
      <Card
        hoverable
        type="inner"
        title="Experience"
        extra={
          <CardExtraContainer>
            {editExperience ? (
              <Button
                data-testid={DATA_TEST_IDS.EXPERIENCE_SAVE}
                onClick={() => {
                  setEditExperience(false);
                }}
              >
                <FormattedMessage {...messages.save} />
              </Button>
            ) : (
              <Button
                data-testid={DATA_TEST_IDS.EXPERIENCE_EDIT}
                onClick={() => {
                  setEditExperience(true);
                }}
              >
                <FormattedMessage {...messages.edit} />
              </Button>
            )}
          </CardExtraContainer>
        }
      >
        {editExperience ? (
          <RichTextEditor
            value={experienceContent}
            onChange={value => {
              setExperienceContent(value);
            }}
          />
        ) : (
          parse(`${isContentEdited(experienceContent)}`)
        )}
      </Card>
      <br />
      <Card
        hoverable
        type="inner"
        title="Education"
        extra={
          <CardExtraContainer>
            {editEducation ? (
              <Button
                data-testid={DATA_TEST_IDS.EDUCATION_SAVE}
                onClick={() => {
                  setEditEducation(false);
                }}
              >
                <FormattedMessage {...messages.save} />
              </Button>
            ) : (
              <Button
                data-testid={DATA_TEST_IDS.EDUCATION_EDIT}
                onClick={() => {
                  setEditEducation(true);
                }}
              >
                <FormattedMessage {...messages.edit} />
              </Button>
            )}
          </CardExtraContainer>
        }
      >
        {editEducation ? (
          <RichTextEditor
            value={educationContent}
            onChange={value => {
              setEducationContent(value);
            }}
          />
        ) : (
          parse(`${isContentEdited(educationContent)}`)
        )}
      </Card>
      <br />
      <Card
        hoverable
        type="inner"
        title="Licenses and Certificates"
        extra={
          <CardExtraContainer>
            {editLicensesAndCertifications ? (
              <Button
                data-testid={DATA_TEST_IDS.LICENSEANDCERTIFICATION_SAVE}
                onClick={() => {
                  setEditLicensesAndCertifications(false);
                }}
              >
                <FormattedMessage {...messages.save} />
              </Button>
            ) : (
              <Button
                data-testid={DATA_TEST_IDS.LICENSEANDCERTIFICATION_EDIT}
                onClick={() => {
                  setEditLicensesAndCertifications(true);
                }}
              >
                <FormattedMessage {...messages.edit} />
              </Button>
            )}
          </CardExtraContainer>
        }
      >
        {editLicensesAndCertifications ? (
          <RichTextEditor
            value={licensesAndCertificationsContent}
            onChange={value => {
              setLicensesAndCertificationsContent(value);
            }}
          />
        ) : (
          parse(`${isContentEdited(licensesAndCertificationsContent)}`)
        )}
      </Card>
      <br />
      <Card hoverable type="inner" title="Upload Resume">
        <FileUpload />
      </Card>
    </div>
  );
}

export default ProfileForm;
