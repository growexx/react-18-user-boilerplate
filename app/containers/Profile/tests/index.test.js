/**
 *
 * Tests for Profile
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { fireEvent, render } from '@testing-library/react';
/**
 * user-event is a companion library for Testing Library
 * that provides more advanced simulation of browser
 * interactions than the built-in fireEvent method
 * https://testing-library.com/docs/ecosystem-user-event/
 */
import userEvent from '@testing-library/user-event';
/**
 * This import adds some helpful assertions.
 * Custom jest matchers to test the state of the DOM
 * https://github.com/testing-library/jest-dom
 */
import '@testing-library/jest-dom';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import { TEST_IDS } from 'components/InlineEdit/stub';
import history from 'utils/history';
import { store } from 'configureStore';
import Profile from '../index';
import { DATA_TEST_IDS } from '../constants';
jest.mock('draft-js/lib/generateRandomKey', () => jest.fn(() => '123'));
jest.mock('components/ImageUpload/Loadable');

const user = userEvent.setup();
let globalStore;

const componentWrapper = () =>
  render(
    <Provider store={globalStore}>
      <IntlProvider locale="en">
        <Router history={history}>
          <Profile />
        </Router>
      </IntlProvider>
    </Provider>,
  );

describe('<Profile />', () => {
  beforeAll(() => {
    globalStore = store;
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('Should click Save Buttons', () => {
    const { getByTestId } = componentWrapper();
    fireEvent.click(getByTestId(DATA_TEST_IDS.ABOUT_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.ABOUT_SAVE));
    expect(getByTestId(DATA_TEST_IDS.ABOUT_EDIT)).toBeTruthy();
    fireEvent.click(getByTestId(DATA_TEST_IDS.EXPERIENCE_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.EXPERIENCE_SAVE));
    expect(getByTestId(DATA_TEST_IDS.EXPERIENCE_EDIT)).toBeTruthy();
    fireEvent.click(getByTestId(DATA_TEST_IDS.EDUCATION_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.EDUCATION_SAVE));
    expect(getByTestId(DATA_TEST_IDS.EDUCATION_EDIT)).toBeTruthy();
    fireEvent.click(getByTestId(DATA_TEST_IDS.LICENSEANDCERTIFICATION_EDIT));
    fireEvent.click(getByTestId(DATA_TEST_IDS.LICENSEANDCERTIFICATION_SAVE));
    expect(
      getByTestId(DATA_TEST_IDS.LICENSEANDCERTIFICATION_EDIT),
    ).toBeTruthy();
  });
  it('Should double click on name and write in input box', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    await user.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    // write on the input
    fireEvent.change(getAllByTestId(TEST_IDS.INPUT_EDIT)[0], {
      target: {
        value: 'testInputChanged',
      },
    });
    // save the input
    fireEvent.click(getAllByTestId(TEST_IDS.SAVE_BUTTON)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)[0]).toBeInTheDocument();
  });
  it('Should double click on designation and write in input box', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    await user.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[1]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    // write on the input
    fireEvent.change(getAllByTestId(TEST_IDS.INPUT_EDIT)[0], {
      target: {
        value: 'testInputChanged',
      },
    });
    // save the input
    fireEvent.click(getAllByTestId(TEST_IDS.SAVE_BUTTON)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)[1]).toBeInTheDocument();
  });
  it('Should double click on location and write in input box', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    await user.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[2]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    // write on the input
    fireEvent.change(getAllByTestId(TEST_IDS.INPUT_EDIT)[0], {
      target: {
        value: 'testInputChanged',
      },
    });
    // save the input
    fireEvent.click(getAllByTestId(TEST_IDS.SAVE_BUTTON)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)[2]).toBeInTheDocument();
  });
  it('Should fire mouse down on other target and input should be disappeared', async () => {
    const { getAllByTestId } = componentWrapper();
    // show the input
    await user.dblClick(getAllByTestId(TEST_IDS.INPUT_VALUE)[2]);
    expect(getAllByTestId(TEST_IDS.INPUT_EDIT)[0]).toBeInTheDocument();
    fireEvent.mouseDown(getAllByTestId(TEST_IDS.INPUT_VALUE)[0]);
    expect(getAllByTestId(TEST_IDS.INPUT_VALUE)).toHaveLength(2);
  });
});
