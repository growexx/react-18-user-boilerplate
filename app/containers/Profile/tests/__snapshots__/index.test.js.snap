// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> Should render and match the snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
}

.c0 .ant-card {
  flex: 1!important;
  cursor: default;
}

.c2 {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.c2 p {
  color: #4d186e;
  font-size: 14px;
  margin-bottom: 0;
}

.c1 .inputValue {
  height: auto;
  padding: 8px;
}

.c1 .inputValue:hover {
  cursor: text;
  background: rgba(9, 30, 66, 0.04);
}

.c1 .placeHolderView {
  color: #808080!important;
}

<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-byeoj0"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="c0"
      >
        <div>
          <div
            class="u-mb-1"
          >
            <div
              class="c1"
            >
              <div
                class="inputValue"
                data-testid="INPUT_VALUE"
              >
                John Doe
              </div>
            </div>
          </div>
          <div
            class="u-mb-1"
          >
            <div
              class="c1"
            >
              <div
                class="inputValue"
                data-testid="INPUT_VALUE"
              >
                Software Engineer at GrowExx
              </div>
            </div>
          </div>
          <div
            class="u-mb-5"
          >
            <div
              class="c1"
            >
              <div
                class="inputValue"
                data-testid="INPUT_VALUE"
              >
                Ahmedabad, India
              </div>
            </div>
          </div>
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner css-dev-only-do-not-override-byeoj0"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  About
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-byeoj0 ant-btn-default"
                      data-testid="About_Edit"
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner css-dev-only-do-not-override-byeoj0"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Experience
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-byeoj0 ant-btn-default"
                      data-testid="Experience_Edit"
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner css-dev-only-do-not-override-byeoj0"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Education
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-byeoj0 ant-btn-default"
                      data-testid="Education_Edit"
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner css-dev-only-do-not-override-byeoj0"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Licenses and Certificates
                </div>
                <div
                  class="ant-card-extra"
                >
                  <div
                    class="c2"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-byeoj0 ant-btn-default"
                      data-testid="LicensesAndCertifications_edit"
                      type="button"
                    >
                      Edit
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
            </div>
          </div>
          <br />
          <div
            class="ant-card ant-card-bordered ant-card-hoverable ant-card-type-inner css-dev-only-do-not-override-byeoj0"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  Upload Resume
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
